### PR TITLE
Several updates

### LINKED ISSUE #14: feature: support for giella-pairs

### LINKED ISSUE #19: define some LEXCFLAGS?
like in
https://github.com/flammie/apertium-nio/commit/e1f449414ed9ca267289d8b5c5230d2ee002cfb3

maybe we should also have some $(UPPERCASE) programs

### LINKED ISSUE #21: should run ./autogen.sh in monolinguals
That way, you can do 

```
apertium-init foo
apertium-init bar
apertium-init foo-bar
cd apertium-foo-bar
./autogen.sh --with-lang1=../apertium-foo --with-lang2=../apertium-bar
make langs
```

### LINKED ISSUE #32: Support bootstrapping pairs where one side is from GT infra
Giellatekno pairs require slightly different makefile stuffs.  

These pairs are both directions GT:

git@github.com:/apertium/apertium-sme-sma.git
git@github.com:/apertium/apertium-sme-smj.git
git@github.com:/apertium/apertium-sme-smn.git

These are one direction GT:

git@github.com:/apertium/apertium-sme-nob.git

### LINKED ISSUE #55: Auto warn when TODOs not replaced on push
cc @ftyers @jonorthwash

### LINKED ISSUE #73: add -f line to test target of bilingual modules in Makefile.am
in the bilingual modules the default test target echos single word to apertium and checks that the translation is expected, but for in-development pairs this produces spurious failures due to apertium adding full stop at the end of the lines, the -f line mode should be used for testing.

### LINKED ISSUE #74: readme's should list common build tools and stuff as requirement
IT's a common problem on apertium IRC channel that people are trying to make apertium languages or pairs without pkg-config on system and it fails with hard to  understand error message from autotools’ guts. The pkg-config etc. should be listed in the readme as requirements.

### LINKED ISSUE #82: Add support for giellatekno language modules

### LINKED ISSUE #90: add udx mapping stuff to language dirs
https://sourceforge.net/p/apertium/mailman/message/37177654/

Probably make something based on the wiki page / the tags that are already in the dix of apertium-init.

### LINKED ISSUE #91: add lexd support
Now that enough people are using lexd, and there are plans to use it even more widely, I think it's time to support it.

There was a regex floating around to convert lexc support in `Makefile.am` to lexd, but it'd be nice if there were an equivalent example file too (`house<n>` or whatever).

### LINKED ISSUE #92: -pe should equal -p -e, not a long form
E.g., https://github.com/apertium/apertium-init/blob/master/main.py#L224 has `parser.add_argument('-pe', '--push-existing-to-github', ...` which is just wrong. There's also `-a1` and others. These should all be `--` instead, or changed to an actual single symbol.

### LINKED ISSUE #93: Newer flake8 unhappy with quotes
`main.py:34:5491: Q003 Change outer quotes to avoid escaping inner quotes` which I assume is because of entries like `'mgo': 'Meta\''`

### LINKED ISSUE #95: validate option combinations
as mentioned in #83 the help text says options can only be used in with monolingual or with pairs but this mostly isn't validated.

### LINKED ISSUE #98: move all instructions in README related to installation to new Installation section near top, before Usage
This issue was automatically made by begiak, Apertium's beloved IRC bot, by the order of firespeaker on #apertium. A human is yet to update the description.

### LINKED ISSUE #99: .deps/spellrelax.hfst should be .deps/$(LANG1).spellrelax.hfst
This issue was automatically made by begiak, Apertium's beloved IRC bot, by the order of firespeaker on #apertium. A human is yet to update the description.

### LINKED ISSUE #100: change to xfst-based spellrelax
This issue was automatically made by begiak, Apertium's beloved IRC bot, by the order of firespeaker on #apertium. A human is yet to update the description.

### LINKED ISSUE #101: vanilla rtx has deprecated indexed blank
This issue was automatically made by begiak, Apertium's beloved IRC bot, by the order of firespeaker on #apertium. A human is yet to update the description.

### PR DESCRIPTION
**Pull request by mr-martian**
* Added support for pairs using Giella modules (closes #14, closes #32, closes #82)
* Added LEXCFLAGS to HFST Makefiles (closes #19)
* Now runs `./autogen.sh` after generating a language module (closes #21)
* Adds a git hook to warn users if they push without updating README TODOs (closes #55)
* Adds `-f line` to `make test` (closes #73)
* Adds mention of autotools etc. to READMEs (closes #74)
* Adds support for lexd (closes #91)
* Removes multi-character short options (closes #92)
* Puts language names and codes in a .tsv file for greater maintainability (continuation of #88 and closes #93)
* Updates language list based on https://iso639-3.sil.org/code_tables/download_tables
* Adds a script to update language list in future
* More thorough validation of CLI arguments (closes #95)
* Leverages conditional replacement code to simplify file inclusion logic (i.e. MANIFEST files)
* Dictionary files now contain links to https://wiki.apertium.org/wiki/List_of_symbols
* Updates main README's installation instructions (closes #98)
* Ensures that `make install` and `python3 setup.py install` both work as expected
* Adds UDX tag mappings (closes #90)
* Removes deprecated blank indexing in .rtx files (closes #101)
* Converts spellrelax from twolc to xfst (closes #100)
* Renames spellrelax transducer (closes #99)

### CODE REVIEW

**Comment by mr-martian** [comment]
I was pretty sure the version needed bumping but wasn't really sure which part. I went with major because `-a1 → --a1`, `-pe → --pe`, and running `./autogen.sh` entail changing the usage instructions.

**Comment by TinoDidriksen** [comment]
Forgot to commit the MANIFEST files? Or forgot to generate them? The build complains they're not there, anyway.

**Comment by mr-martian** [comment]
Crap, I did forget to commit the MANIFEST files. I also realized that it's currently not possible to use all the new Giella code... I'll go fix those.

**Comment by mr-martian** [comment]
`MANIFEST` was caught by the Python `.gitignore` and I didn't notice when I was committing. I've changed it to `MANIFEST.txt`.

File: main.py [hunk 1]
```diff
@@ -216,20 +206,22 @@ def create_github_repository():  # type: () -> http.client.HTTPResponse
 
 
 def main(cli_args):  # type: (List[str]) -> None
-    parser = argparse.ArgumentParser(description='Bootstrap an Apertium language module/pair')
+    parser = argparse.ArgumentParser(description='Bootstrap an Apertium language module/pair', allow_abbrev=False)
```
Comment (TinoDidriksen) [inline main.py:209]: https://github.com/apertium/apertium-init/issues/92#issuecomment-759262814

**Review by TinoDidriksen — APPROVED** [review]

**Comment by jonorthwash** [comment]
> Updates language list based on https://iso639-3.sil.org/code_tables/download_tables

Does this mean that there's a build-system triggered way for the updates to happen, or that there's some method that needs to be triggered manually, or something else?

**Comment by mr-martian** [comment]
I did it by hand, though the conversion process is essentially just deleting a few columns, so it could be automated without too much trouble. Adding a manual update script would probably be a good idea.

**Review by TinoDidriksen — APPROVED** [review]

**Comment by jonorthwash** [comment]
`make` doesn't seem to do the right thing, and `setup.py install` doesn't work without first `make apertium_init.py`

**Comment by jonorthwash** [comment]
If you just run the installed version, it crashes instead of printing help:
```
$ apertium-init
Traceback (most recent call last):
  File "/usr/local/bin/apertium-init", line 33, in <module>
    sys.exit(load_entry_point('apertium-init==3.0.0', 'console_scripts', 'apertium-init')())
TypeError: main() missing 1 required positional argument: 'cli_args'
```

**Comment by mr-martian** [comment]
There are somethings in the Makefile which were written when `apertium-init.py` always existed and I missed them in #88. `make install` now works for me.

**Comment by jonorthwash** [comment]
> There are somethings in the Makefile which were written when `apertium-init.py` always existed and I missed them in #88. `make install` now works for me.

So `make; sudo make install` works, but so does `make; sudo python3 setup.py install`.  Which is the preferred way / is there a difference / what happens if you do them both?

**Comment by mr-martian** [comment]
Running `setup.py` makes it possible to do `import apertium_init`. If you run both then which one you get when you type `apertium-init` on the command line will depend on your path. Otherwise after 895c922 they should be equivalent.

**Comment by flammie** [comment]
I checked the hfst / autotools parts quickly and it lgtm but haven't tested, I could actually try one new pair soon...

**Review by TinoDidriksen — APPROVED** [review]

**Pull request merged**